instagram/src/components/AppContainer/AppContainer.js

```diff
@@ -0,0 +1,38 @@
+import React, { Component } from "react";
+
+import data from "../../data/dummy-data";
+
+import SearchBar from "./SearchBar/SearchBar";
+import PostContainer from "./PostContainer/PostContainer";
+
+class AppContainer extends Component {
+  state = {
+    posts: []
+  };
+
+  componentDidMount() {
+    this.setState({ posts: [...data] });
+  }
+
+  searchPosts = e => {
+    this.setState({
+      posts: data.filter(post => post.username.includes(e.target.value))
```

Comment:
Keep in mind that your logic doesn't apply to uppercased letters. Might want to do that. 

post.username.toLowerCase().includes(e.target.value.toLowerCase())